test/integration/targets/azure_rm_localnetworkgateway/tasks/main.yml

```diff
@@ -0,0 +1,125 @@
+- name: Prepare dummy parameter values
+  set_fact:
+    gateway_ip_address: 1.2.3.4
+    peer_asn: 65000
+    bgp_peering_address: 169.254.53.201
+    rpfx: "{{ resource_group | hash('md5') | truncate(7, True, '') }}{{ 1000 | random }}"
+
+- name: Create local network gateway
+  azure_rm_localnetworkgateway:
+    name: lgw{{ rpfx }}
+    local_address_prefixes:
+      - 10.1.0.0/16
+      - 172.100.0.0/16
+    gateway_ip_address: "{{ gateway_ip_address }}"
+    tags:
+      testing: testing
+      delete: on-exit
+    resource_group: "{{ resource_group }}"
+  register: output
+
+- assert:
+    that: output.changed
+
+- name: Create local network gateway(check-mode)
+  azure_rm_localnetworkgateway:
+    name: lgw{{ rpfx }}
+    local_address_prefixes:
+      - 10.1.0.0/16
+      - 172.100.0.0/16
+    gateway_ip_address: "{{ gateway_ip_address }}"
+    tags:
+      testing: testing
+      delete: on-exit
+    resource_group: "{{ resource_group }}"
+  check_mode: yes
+  register: output
+
+- assert:
+    that: output.changed
```

Comment:
this is idempotent test,  and output.change should be false if nothing changed.

Reply:
@yungezz  Please correct me if I am wrong, but as this is a 'check_mode' test, it should always return 'true' right? That's the reason for asserting on output.changed.
If the assumption is wrong, I would be happy to change the implementation. Please check

Reply:
@Madhura-CSI Thanks for your contribution, Perhaps you can adjust your YML order to make it easier for other maintainers to understand, As this order: 

1. Create local network gateway --- check mode

2. Create local network gateway

3. Re-create local network gateway ----  idempotent test

4. Update local network gateway with address prefixes and tags

5.  Gather facts or catch required argument gateway_ip_address

Reply:
@Madhura-CSI  you're right. i looked at wrong test.